### PR TITLE
Migrate allowed unsafe sysctls to kubelet config group

### LINKED ISSUE #61669: Migrate Kubelet --experimental-allowed-unsafe-sysctls to kubelet.config.k8s.io or remove the flag
Flag name: `experimental-allowed-unsafe-sysctls`

Help text:

Comma-separated whitelist of unsafe sysctls or unsafe sysctl patterns (ending in *). Use these at your own risk.

---

This is part of migrating the Kubelet command-line to a Kubernetes-style API.

The `--experimental-allowed-unsafe-sysctls` flag should either be migrated to the Kubelet's `kubelet.config.k8s.io`
API group, or simply removed from the Kubelet.

If this could be considered an instance-specific flag, or a descriptor of local topology
managed by the Kubelet, see: https://github.com/kubernetes/kubernetes/issues/61647.

If this flag is only registered in os-specific builds, see: https://github.com/kubernetes/kubernetes/issues/61649.

As `--experimental-allowed-unsafe-sysctls` is an alpha/experimental flag, the feature it configures must either be
feature-gated, or graduated from alpha/experimental status prior to the migration.

@sig-node-pr-reviews @sig-node-api-reviews

/assign @mtaufen
/sig node
/kind feature
/priority important-soon
/milestone v1.11
/status approved-for-milestone

### PR DESCRIPTION
**Pull request by ingvagabund**
**What this PR does / why we need it**:
Get closer to dynamic Kubelet configuration

**Which issue(s) this PR fixes**
Fixes #61669 

**Special notes for your reviewer**:
Life without a beer is complicated

**Release note**:

```release-note
TBD
```

Built on top of https://github.com/kubernetes/kubernetes/pull/63717. Once the PR gets merged, I will rebase.

### CODE REVIEW

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -544,6 +539,8 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 	fs.StringVar(&c.ContainerLogMaxSize, "container-log-max-size", c.ContainerLogMaxSize, "<Warning: Alpha feature> Set the maximum size (e.g. 10Mi) of container log file before it is rotated. This flag can only be used with --container-runtime=remote.")
 	fs.Int32Var(&c.ContainerLogMaxFiles, "container-log-max-files", c.ContainerLogMaxFiles, "<Warning: Alpha feature> Set the maximum number of container log files that can be present for a container. The number must be >= 2. This flag can only be used with --container-runtime=remote.")
 
+	fs.StringSliceVar(&c.AllowedUnsafeSysctls, "allowed-unsafe-sysctls", c.AllowedUnsafeSysctls, "Comma-separated whitelist of unsafe sysctls or unsafe sysctl patterns (ending in *). Use these at your own risk. Sysctls feature gate is enabled by default.")
```
Comment (mtaufen) [inline cmd/kubelet/app/options/options.go]: What is the status of this feature? (alpha, beta, GA?)
Which feature gate applies to it?

"Use these at your own risk" makes me a little nervous moving the flag into a beta API... but that's potentially just an old comment that is no longer relevant.

Comment (ingvagabund) [reply cmd/kubelet/app/options/options.go]: Going to Beta (PR almost merged). And yes, the `"Use these at your own risk"` can be removed. I will update the PR once rebased.

Comment (ingvagabund) [reply cmd/kubelet/app/options/options.go]: The `"Use these at your own risk" ` is actually intentional. By default, no unsafe sysctl is allowed. Though, there are some applications that require (or work better) with some unsafe sysctls allowed. So it's up to an admin to (re)consider which unsafe sysctls are "ok" at his own risk.

**Comment by ingvagabund** [comment]
@mtaufen this PR will sit here for a while given it will not get into 1.11 release. Is there any plan deadline to when all the flags are supposed to be migrated? Until 1.12?

**Comment by mtaufen** [comment]
We don't have an official deadline, but sooner is better in almost all cases.

File: pkg/kubelet/apis/kubeletconfig/v1beta1/types.go [hunk 2]
```diff
@@ -634,6 +634,9 @@ type KubeletConfiguration struct {
 	// Default: "Watching"
 	// +optional
 	ConfigMapAndSecretChangeDetectionStrategy ResourceChangeDetectionStrategy `json:"configMapAndSecretChangeDetectionStrategy,omitempty"`
+	// Whitelist of unsafe sysctls or sysctl patterns (ending in *).
+	// +optional
+	AllowedUnsafeSysctls []string
```
Comment (sjenning) [inline pkg/kubelet/apis/kubeletconfig/v1beta1/types.go]: needs json tag

**Comment by rphillips** [comment]
/reopen

This is still needed.

**Comment by sjenning** [comment]
we don't need everything in here; just the converting the flag to a field on KubeletConfguration.  i'm working on a PR.

**Pull request closed without merging**